### PR TITLE
added docu for options for file access

### PR DESCRIPTION
**Pull request by behrica**

### CODE REVIEW

File: docs/kubernetes_azureStorage.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+
```
Comment (anthony-khong) [inline docs/kubernetes_azureStorage.md:1]: Can we change the title to use the underscore convention? `azureStorage` -> `azure_storage`.

File: docs/kubernetes_azureStorage.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+```bash
+docker build -t genidemo18w.azurecr.io/geni .
+
+az acr login --resource-group geni-azure-demo --name genidemo18w
+docker push genidemo18w.azurecr.io/geni
+```
+
+# create namespaces
```
Comment (anthony-khong) [inline docs/kubernetes_azureStorage.md]: Capitalise: `create` -> `Create`

File: docs/kubernetes_azureStorage.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+```clojure
+(require '[zero-one.geni.core :as g])
+
+;; This should be the first function executed in the repl.
+
+(g/create-spark-session
+ {:app-name "my-app"
+  :log-level "INFO" ;; default is WARN
+  :configs
+  {:spark.master "k8s://https://kubernetes.default.svc" ;;  might differ for you, its the output of kubecl cluster-info
```
Comment (anthony-khong) [inline docs/kubernetes_azureStorage.md:267]: Can we change the comments to text that explains the decisions to include certain configs and the reasons why you'd need to change the default ones?

File: docs/kubernetes_basic.md [hunk 4]
```diff
@@ -132,3 +132,29 @@ my-app-a9f44174fa720b22-exec-3   1/1     Running   0          89s
...
+## Reading files into Spark on Kubernetes
+
+We have a lot of different options for reading files into Spark in a Kubernetes setup.
+The main requirement is, that the files can be addressed under a common URL in the driver and all workers.
+
+In Geni all file reading happens via functions of form  `g/read_xxx_!`. We can use different (eventually remote) file sytems by using different forms of the URL (specifically by using different prefixes).
+
+The reading of  files by Geni can be achieved in 4 different ways:
+
+0. (Just copy the data files into the same path in all the Kubernetes pods and the driver machine. This is very adhoc and likely not suitable in most scenarios, as the data files would be lost at the end of the REPL session due to pods being terminated.)
```
Comment (anthony-khong) [inline docs/kubernetes_basic.md:144]: Let's leave the parentheses out. The points being made here are actually very important.

File: docs/kubernetes_basic.md [hunk 5]
```diff
@@ -132,3 +132,29 @@ my-app-a9f44174fa720b22-exec-3   1/1     Running   0          89s
...
+ 
+1. (Add data files to custom  Docker images used for Driver and Worker nodes. This is rather inconvenient as well, as any change in the data requires different Docker images)
```
Comment (anthony-khong) [inline docs/kubernetes_basic.md:146]: Same here, let's remove the parens.

File: docs/kubernetes_basic.md [hunk 6]
```diff
@@ -132,3 +132,29 @@ my-app-a9f44174fa720b22-exec-3   1/1     Running   0          89s
...
+ 
+2. Mount remote filesystem into the pod on operating system level. This makes them accessible as local files. Example: [blobfuse](https://github.com/Azure/azure-storage-fuse). This has direct support for Kubenetes as well: [kubernetes volume driver](https://github.com/Azure/kubernetes-volume-drivers) (This is then the same as 3.)
```
Comment (anthony-khong) [inline docs/kubernetes_basic.md:148]: `Example: [blobfuse](https://github.com/Azure/azure-storage-fuse)`
->
`See for example [blobfuse](https://github.com/Azure/azure-storage-fuse)`.

`This has direct support for Kubenetes as well: [kubernetes volume driver](https://github.com/Azure/kubernetes-volume-drivers) (This is then the same as 3.)`
->
`This has direct support for Kubenetes as well through [kubernetes volume driver](https://github.com/Azure/kubernetes-volume-drivers), which is then the same as 3.`

File: docs/kubernetes_basic.md [hunk 7]
```diff
@@ -132,3 +132,29 @@ my-app-a9f44174fa720b22-exec-3   1/1     Running   0          89s
...
+ 
+3. __Mount Kubernetes data volumes into pods:__ Using a __local__ file path __and__ make sure that these path exist and work in all workers __and__ the driver. In Kubernets/Docker we can do this by __mounting__ (eventually remote) filesystems into the pods at certain location. Doing this, the files would look like local files and urls would not have a specific prefix, and be for example `(g/read "/data/test.csv")` This approach can eventually become easier, if we make sure, that the driver is running as well inside Kubernetes. (In the minikube setup above, the driver runs __outside__ Kubernetes)
```
Comment (anthony-khong) [inline docs/kubernetes_basic.md:150]: I think it's nice to make the gist of the point bold. Should we do it to points 0, 1 and 2 as well?

"easier, if" -> "easier if"

And also, let's remove the final parens.

File: docs/kubernetes_azureStorage.md (hunk 1 above)
Comment (behrica) [reply docs/kubernetes_azureStorage.md:1]: I added the file kubernetes_ayureStorage.md by mistake already in this pull request...

This is work in progress.

I will address all you suggestions in file kubernets_basic and then do a new PR

**Comment by behrica** [comment]
Replaced by #269

**Pull request closed without merging**